pywisetransfer/deprecation.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+import warnings
+
+
+def deprecated(message, *args, **kwargs):
+    """Decorator to indicate that a method is deprecated, with an optional
+    message to emit in the warning.
+
+    Written following guidance from:
+    https://blog.miguelgrinberg.com/post/the-ultimate-guide-to-python-decorators-part-iii-decorators-with-arguments
+    """
+
+    # General-case: handle decorators with arguments; @deprecated(message='foo')
+    def decorator(f):
```

Comment:
@marksteward not important if you're busy, but are you familiar enough with Python decorator magic to help me avoid the object signatures for items decorated as `@deprecated` from looking.. spammy?

```pyi
>>> import pywisetransfer
>>> client = pywisetransfer.Client(api_key='testing')
>>> client.borderless_accounts.statement
<bound method deprecated.<locals>.decorator.<locals>.deprecation_notice of <pywisetransfer.borderless_account.BorderlessAccount object at <0xblah>>
```

Reply:
(please ignore: this has been resolved)